src/rgw/rgw_common.h

```diff
@@ -1102,6 +1102,31 @@ enum RGWBucketIndexType {
   RGWBIType_Indexless = 1,
 };
 
+inline ostream& operator<<(ostream& out, const RGWBucketIndexType &index_type) 
+{
+  string str;  
+
+  switch (index_type) {
+    case RGWBIType_Normal:
+    {
+      str = "Normal";
+      break;
+    }
+    case RGWBIType_Indexless:
+    {
+      str = "Indexless";
+      break;
+    }
+    default:
+    {
+      str = "Unknown";
+      break;
+    }
+  }
+  
+  return out << str;
+}
+
```

Comment:
can you try to shorten this function a bit? the temporary `string str` shouldn't be necessary. something like:
```c++
inline std::ostream& operator<<(std::ostream& out, RGWBucketIndexType index_type)
{
  switch (index_type) {
  case RGWBIType_Normal:
    return out << "Normal";
  case RGWBIType_Indexless:
    return out << "Indexless";
  default:
    return out << "Unknown";
  }
}
```